src/main/resources/bullet_kafka_defaults.yaml

```diff
@@ -1,33 +1,67 @@
-# These are required Kafka settings for consumers and producers
-# For more details see https://kafka.apache.org/documentation/#configuration
+# These are required Kafka settings for consumers and producers.
+# For more details see https://kafka.apache.org/documentation/#configuration.
 
-# Common Properties
+# Common Kafka Properties
 bullet.pubsub.kafka.bootstrap.servers: "localhost:9091"
+bullet.pubsub.kafka.connections.max.idle.ms: "-1"
+
+# Kafka Producer properties
+# These are passed to KafkaProducer. You can add other Kafka properties by prefixing the property with "bullet.pubsub.kafka.producer".
+bullet.pubsub.kafka.producer.acks: "all"
+bullet.pubsub.kafka.producer.retries: "5"
+bullet.pubsub.kafka.producer.batch.size: "65536"
+bullet.pubsub.kafka.producer.linger.ms: "5"
+bullet.pubsub.kafka.producer.buffer.memory: "33554432"
+bullet.pubsub.kafka.producer.request.timeout.ms: "3000"
+bullet.pubsub.kafka.producer.key.serializer: "org.apache.kafka.common.serialization.StringSerializer"
+bullet.pubsub.kafka.producer.value.serializer: "org.apache.kafka.common.serialization.ByteArraySerializer"
+bullet.pubsub.kafka.producer.max.block.ms: "50000"
+
+# Kafka Consumer properties
+# These are passed to KafkaConsumer. You can add other Kafka properties by prefixing the property with "bullet.pubsub.kafka.consumer".
+bullet.pubsub.kafka.consumer.group.id: "bullet-query-consumer"
+bullet.pubsub.kafka.consumer.heartbeat.interval.ms: "3000"
+bullet.pubsub.kafka.consumer.enable.auto.commit: "true"
+bullet.pubsub.kafka.consumer.auto.commit.interval.ms: "1000"
+bullet.pubsub.kafka.consumer.session.timeout.ms: "30000"
+bullet.pubsub.kafka.consumer.max.poll.records: "50"
+bullet.pubsub.kafka.consumer.key.deserializer: "org.apache.kafka.common.serialization.StringDeserializer"
+bullet.pubsub.kafka.consumer.value.deserializer: "org.apache.kafka.common.serialization.ByteArrayDeserializer"
+bullet.pubsub.kafka.consumer.fetch.max.wait.ms: "500"
+# These following are optional but must be greater than bullet.pubsub.sleep-ms in the web-service configuration if
+# "bullet.pubsub.context.name" is QUERY_SUBMISSION.
+bullet.pubsub.kafka.consumer.max.poll.interval.ms: "30000"
+bullet.pubsub.kafka.consumer.request.timeout.ms: "35000"
+
+# Kafka PubSub properties
+# The number of messages that can received before at least one commit is needed.
+bullet.pubsub.kafka.consumer.max.uncommitted.messages: 50
```

Comment:
This doesn't need to be a string?

Reply:
This is a Bullet config so I can keep proper types. Kafka params are better off being Strings since Kafka casts. Otherwise, they need to be the exact type (it doesn't do Longs to Integer etc). That being said, this param is confusing because it has the same prefix as the Kafka consumer so I'll remove the kafka from the prefix.